src/test/java/com/sumologic/log4j/SumoLogicAppenderTest.java

```diff
@@ -105,4 +105,10 @@ public void testMessagesWithoutMetadata() throws Exception {
             assertEquals("log4j2-appender", request.getHeaders().getFirst("X-Sumo-Client"));
         }
     }
+
+    @Test
+    public void setupAppenderWithoutRequiredFields() {
+        SumoLogicAppender appender = SumoLogicAppender.newBuilder().build();
+        assertNull(appender);
```

Comment:
Why would it be null? I fail to understand.

Reply:
Because builder is using the createAppender function under the hood and it returns null if validation fails + logs ERROR message.

Reply:
OK. I understand now.
Now I even get why would you ask if we need a test like this.